[website] Clarify grandfathering policy (#45265)
mui/material-ui: docs/pages/blog/mui-x-sep-2024-price-update.md

We value our loyal customers and want to ensure a smooth transition: current customers will be grandfathered into their existing plans.

> This means you can renew your licenses, purchase additional licenses, under the legacy pricing for as long as you wish.

Reviewer: Well you can do it now too, so future tense isn't super relevant. If you opt for legacy plan though instead of legacy pricing, then "under" fits better grammatically.

```suggestion
This means you will be able to renew your licenses or purchase additional ones under the legacy plan for as long as you wish.
```

However, we want to ensure that our valued early customers continue to enjoy the benefits of an earlier purchase.
Current customers will still benefit from a 20% renewal discount (vs. 25% early bird discount before).

> This means you will be able to renew your licenses, purchase additional licenses at a 7% price increase compare to before while gaining access to new features.

Reviewer: ```suggestion
This means you can renew your licenses or purchase additional licenses at a 7% price increase while gaining access to new features.
```

Author: I guess we should keep the past tense, as it was right about the future.

```suggestion
This means you will be able to renew your licenses or purchase additional licenses at a 7% price increase while gaining access to new features.
```

> You'll receive the discount per email, you can also receive it by reaching out to our [sales team](mailto:sales@mui.com).

Reviewer: ```suggestion
You'll receive the discount via email, you can also request it by contacting our [sales team](mailto:sales@mui.com).
```
